src/location.js

```diff
@@ -37,12 +39,16 @@ const Location = ({homeUrl, children}) => {
     // Set the location state to a parsed url and a compiled url.
     const [ parsedUrl, setParsedUrl ] = useState(parseJsonApiUrl(homeUrl));
     const [ locationUrl, setLocationUrl ] = useState(compileJsonApiUrl(parsedUrl));
+    const [ document, setDocument ] = useState({});
 
     // Takes a single query parameter and updates the parsed url.
     const updateQuery = (param) => setParsedUrl(Object.assign({}, parsedUrl, {query: Object.assign({}, parsedUrl.query, param)}));
 
     // If the parsed url is updated, compile it and update the location url.
     useEffect(() => setLocationUrl(compileJsonApiUrl(parsedUrl)), [parsedUrl]);
+    useEffect(() => {
```

Comment:
For some reason I was thinking you could only have one `useEffect` -but it's just a function, of course you can call it for each separate requirement!